etlman/projects/views.py

```diff
@@ -27,7 +30,29 @@ def step_form_upsert_view(request, pk=None):
             return HttpResponseRedirect(
                 reverse("projects:step_form_upsert", args=[saved_obj.id])
             )
-    else:  # GET
+    else:  # GET request
         form = StepForm(instance=loaded_obj)
     context = {"form": form}
     return render(request, "projects/step_form.html", context)
+
+
+@authorize(any_authorized)  # any logged in user
+def new_project_wizard_view(request):
+    username = request.user.username
+
+    # Form functionality
+    if request.method == "POST":
+        form = ProjectForm(request.POST)
+        if form.is_valid():
+            saved_project = form.save()
+            # New projects require the logged user as a collaborator
+            # Here we assigned the current user as the "collaborator".
+            Collaborator.objects.create(
+                user=request.user,
+                role="admin",
+                project=saved_project,
+            )
+        return HttpResponseRedirect(reverse("home"))
```

Comment:
This redirect should be indented once more (only redirect if the form was valid).

It might be nice to have a 'success' message like the other view, too. You can use substitution in your messages enum to include the project name, e.g.:

```
class MessagesEnum:
    # ...
    PROJECT_CREATED = "Project {name} added successfully"
```

```
def new_project_wizard_view(request):
    # ...
            messages.add_message(
                request,
                messages.SUCCESS,
                MessagesEnum.PROJECT_CREATED.format(name=saved_project.name)
            )
```